v3: panic when re-generating sandstorm schema

Just tried re-generating the schema in go.sandstorm using the v3 branch, and got this:

```
panic: Which() != text

goroutine 1 [running]:
zombiezen.com/go/capnproto2/internal/schema.Value.Text(0xc4200ff4a0, 0x1000002138, 0x1, 0x38, 0x0, 0xc4200ff4a0, 0x1000002138, 0x1, 0x38)
	/home/isd/go/src/zombiezen.com/go/capnproto2/internal/schema/schema.capnp.go:2157 +0x1f5
main.parseAnnotations(0xc4200ff4a0, 0x100002120, 0x200000008, 0x3a, 0x1, 0xc4200ff4a0)
	/home/isd/go/src/zombiezen.com/go/capnproto2/capnpc-go/nodes.go:164 +0x22b
main.makeEnumval(0xc420123f10, 0x1, 0xc4200ff4a0, 0x800001f58, 0x2, 0x3b, 0x1, 0x0, 0x0, 0x0, ...)
	/home/isd/go/src/zombiezen.com/go/capnproto2/capnpc-go/nodes.go:81 +0x157
main.(*generator).defineEnum(0xc42012b0a0, 0xc420123f10, 0x6, 0x3d)
	/home/isd/go/src/zombiezen.com/go/capnproto2/capnpc-go/capnpc-go.go:251 +0x3cf
main.(*generator).defineFile(0xc42012b0a0, 0xc4200ff440, 0x10101)
	/home/isd/go/src/zombiezen.com/go/capnproto2/capnpc-go/capnpc-go.go:1104 +0x29f
main.generateFile(0xc420120060, 0x800000030, 0x2, 0x3d, 0x1, 0xc4200ff440, 0xc420010101, 0x800000030, 0x2)
	/home/isd/go/src/zombiezen.com/go/capnproto2/capnpc-go/capnpc-go.go:1131 +0x12b
main.main()
	/home/isd/go/src/zombiezen.com/go/capnproto2/capnpc-go/capnpc-go.go:1192 +0x6b4
go: plugin failed: exit code 2
```

Steps to reproduce:

* Check out the master branch of go.sandstorm, and make sure capnpc-go is the one from the v3 branch
* cd `$go_sandstorm/capnp`
* `./gen.sh clean-gen`
* `./gen.sh compile`

--- a/capnpc-go/nodes.go
+++ b/capnpc-go/nodes.go
@@ -161,21 +161,20 @@ func parseAnnotations(list schema.Annotation_List) *annotations {
 	for i, n := 0, list.Len(); i < n; i++ {
 		a := list.At(i)
 		val, _ := a.Value()
-		text, _ := val.Text()
 		switch a.Id() {
 		case 0xc58ad6bd519f935e: // $doc
-			ann.Doc = text
+			ann.Doc, _ = val.Text()
 		case 0xbea97f1023792be0: // $package
-			ann.Package = text
+			ann.Package, _ = val.Text()
 		case 0xe130b601260e44b5: // $import
-			ann.Import = text
+			ann.Import, _ = val.Text()
 		case 0xa574b41924caefc7: // $tag
 			ann.TagType = customTag
-			ann.CustomTag = text
+			ann.CustomTag, _ = val.Text()
 		case 0xc8768679ec52e012: // $notag
 			ann.TagType = noTag
 		case 0xc2b96012172f8df1: // $name
-			ann.Name = text
+			ann.Name, _ = val.Text()
 		}
 	}
 	return ann

